internal/core/ebook.go

```diff
@@ -0,0 +1,238 @@
+  <rootfiles>
+	<rootfile full-path="OEBPS/content.opf" media-type="application/oebps-package+xml"/>
+  </rootfiles>
+</container>`))
+
+	contentOpfWriter, err := epubWriter.Create("OEBPS/content.opf")
+	if err != nil {
+		return book, true, fmt.Errorf("can't create content.opf")
+	}
+	contentOpfWriter.Write([]byte(`<?xml version="1.0" encoding="UTF-8"?>
+<package xmlns="http://www.idpf.org/2007/opf" version="2.0" unique-identifier="BookId">
+  <metadata>
+    <dc:title>` + book.Title + `</dc:title>
+    <dc:language>en</dc:language>
+    <dc:identifier id="BookId">urn:uuid:12345678-1234-5678-1234-567812345678</dc:identifier>
+  </metadata>
+  <manifest>
+    <item id="ncx" href="toc.ncx" media-type="application/x-dtbncx+xml"/>
+    <item id="content" href="content.html" media-type="application/xhtml+xml"/>
+	<item id="id" href="../style.css" media-type="text/css"/>
+  </manifest>
+  <spine toc="ncx">
+    <itemref idref="content"/>
+  </spine>
+</package>`))
+
+	// Create the style.css file
+	styleWriter, err := epubWriter.Create("style.css")
+	if err != nil {
+		return book, true, fmt.Errorf("can't create content.xml")
+	}
+	styleWriter.Write([]byte(`content {
+	display: block;
+	font-size: 1em;
+	line-height: 1.2;
+	padding-left: 0;
+	padding-right: 0;
+	text-align: justify;
+	margin: 0 5pt
+}
+img {
+  	margin: auto;
+  	display: block;
+}`))
+	// Create the toc.ncx file
+	tocNcxWriter, err := epubWriter.Create("OEBPS/toc.ncx")
+	if err != nil {
+		return book, true, fmt.Errorf("can't create toc.ncx")
+	}
+	tocNcxWriter.Write([]byte(`<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE ncx PUBLIC "-//NISO//DTD ncx 2005-1//EN"
+  "http://www.daisy.org/z3986/2005/ncx-2005-1.dtd">
+<ncx xmlns="http://www.daisy.org/z3986/2005/ncx/" version="2005-1">
+  <head>
+    <meta name="dtb:uid" content="urn:uuid:12345678-1234-5678-1234-567812345678"/>
+    <meta name="dtb:depth" content="1"/>
+    <meta name="dtb:totalPageCount" content="0"/>
+    <meta name="dtb:maxPageNumber" content="0"/>
+  </head>
+  <docTitle>
+    <text>` + book.Title + `</text>
+  </docTitle>
+  <navMap>
+    <navPoint id="navPoint-1" playOrder="1">
+      <navLabel>
+        <text >` + book.Title + `</text>
+      </navLabel>
+      <content src="content.html"/>
+    </navPoint>
+  </navMap>
+</ncx>`))
+
+	containerWriter.Write([]byte(`<?xml version="1.0" encoding="UTF-8"?>
+<container version="1.0" xmlns="urn:oasis:names:tc:opendocument:xmlns:container">
+  <rootfiles>
+    <rootfile full-path="OEBPS/content.opf" media-type="application/oebps-package+xml"/>
+  </rootfiles>
+</container>`))
+
+	// get list of images tag in html
+	imageList, _ := getImages(book.HTML)
+	imgRegex := regexp.MustCompile(`<img.*?src="([^"]*)".*?>`)
+
+	// Create a set to store unique image URLs
+	imageSet := make(map[string]bool)
+
+	// Download image in html file and generate new html
+	html = book.HTML
+	for _, match := range imgRegex.FindAllStringSubmatch(book.HTML, -1) {
```

Comment:
In the future, we should try to aim and use the content we already have stored, but that could be complicated and we need to migrate from warc to obelisk (and potentially allow other archival generators) so this is just a general comment and not that you need to perform any action, just so you have that in mind.